src/main/java/uk/gov/digital/ho/hocs/info/api/TopicService.java

```diff
@@ -20,31 +25,229 @@
+                    auditClient.deleteParentTopicAudit(parentTopic);
+                }
+            }
+            parentTopic.setActive(false);
+            parentTopicRepository.save(parentTopic);
+            log.info("Deleted parent topic: {}", parentTopic.getDisplayName());
+
+        }
+    }
+
+    public void deleteTopic(UUID topicUUID) {
+
+        log.debug("Deleting topic: {}", topicUUID);
+        Topic topic = topicRepository.findTopicByUUID(topicUUID);
+        if (topic == null){
+            log.debug("Unable to delete  topic, UUID: {} does not exist", topicUUID);
+            throw new ApplicationExceptions.EntityNotFoundException
+                    ("Unable to delete topic, UUID: {} does not exist", topicUUID);
+        } else {
+            topic.setActive(false);
+            topicRepository.save(topic);
+            log.info("Deleted topic: {}", topic.getDisplayName());
+            auditClient.deleteTopicAudit(topic);
+        }
+    }
+
+    public void reactivateParentTopic(UUID parentTopicUUID) {
+
+        log.debug("Reactivating parent topic: {}", parentTopicUUID);
+        ParentTopic parentTopic = parentTopicRepository.findByUuid(parentTopicUUID);
+        if (parentTopic == null){
+            throw new ApplicationExceptions.EntityNotFoundException("Parent topic with UUID {} does not exist", parentTopicUUID.toString());
+        }
+        if (parentTopic.getActive() == true) {
+            log.debug("Parent topic is already active");
+        } else {
+            parentTopic.setActive(true);
+            parentTopicRepository.save(parentTopic);
+            log.info("Activated parent topic: {}", parentTopicUUID);
+            auditClient.reactivateParentTopicAudit(parentTopic);
+        }
+    }
+
+    public void reactivateTopic(UUID topicUUID) {
+
+        log.debug("Reactivating topic: {}", topicUUID);
+        Topic topic = topicRepository.findTopicByUUID(topicUUID);
+        if (topic == null){
+            throw new ApplicationExceptions.EntityNotFoundException("Topic with UUID {} does not exist", topicUUID.toString());
+        }
+        if (topic.getActive() == true) {
+            log.debug("Topic is already active");
+        } else {
+            ParentTopic parentTopic = parentTopicRepository.findByUuid(topic.getParentTopic());
+            if(parentTopic.getActive() == false){
+                log.debug("Unable to activate topic: {}, parent topic is inactive", topicUUID);
+                throw new ApplicationExceptions.TopicUpdateException("Unable to activate topic, parent topic is inactive");
+            } else {
+                topic.setActive(true);
+                topicRepository.save(topic);
+                log.info("Activated topic: {}", topicUUID);
+                auditClient.reactivateTopicAudit(topic);
+            }
+        }
+    }
+
+    public void updateTopicName(String displayName, UUID topicUUID) {
```

Comment:
Check this one with Ed, I don't think it should be possible to rename a topic, they should be created or deleted only.

Reply:
Yep, just checked and I mixed up the requirement. Now I need to add a link to teams!